superset-frontend/src/views/CRUD/welcome/ChartTable.tsx

```diff
@@ -0,0 +1,170 @@
+import { useListViewResource, useChartEditModal } from 'src/views/CRUD/hooks';
+import withToasts from 'src/messageToasts/enhancers/withToasts';
+import PropertiesModal from 'src/explore/components/PropertiesModal';
+import { User } from 'src/types/bootstrapTypes';
+import Icon from 'src/components/Icon';
+import ChartCard from 'src/views/CRUD/chart/ChartCard';
+import Chart from 'src/types/Chart';
+import SubMenu from 'src/components/Menu/SubMenu';
+import EmptyState from './EmptyState';
+import { CardContainer, IconContainer } from '../utils';
+
+const PAGE_SIZE = 3;
+
+interface ChartTableProps {
+  addDangerToast: (message: string) => void;
+  addSuccessToast: (message: string) => void;
+  search: string;
+  chartFilter?: string;
+  user?: User;
+}
+
+function ChartTable({
+  user,
+  addDangerToast,
+  addSuccessToast,
+}: ChartTableProps) {
+  const {
+    state: { loading, resourceCollection: charts, bulkSelectEnabled },
+    setResourceCollection: setCharts,
+    hasPerm,
+    refreshData,
+    fetchData,
+  } = useListViewResource<Chart>('chart', t('chart'), addDangerToast);
+
+  const {
+    sliceCurrentlyEditing,
+    openChartEditModal,
+    handleChartUpdated,
+    closeChartEditModal,
+  } = useChartEditModal(setCharts, charts);
+
+  const [chartFilter, setChartFilter] = useState('Favorite');
+
+  const getFilters = () => {
+    const filters = [];
+
+    if (chartFilter === 'Mine') {
+      filters.push({
+        id: 'created_by',
+        operator: 'rel_o_m',
+        value: `${user?.userId}`,
+      });
+    } else {
+      filters.push({
+        id: 'id',
+        operator: 'chart_is_fav',
+        value: true,
+      });
+    }
+    return filters;
+  };
+
+  useEffect(() => {
+    fetchData({
+      pageIndex: 0,
+      pageSize: PAGE_SIZE,
+      sortBy: [
+        {
+          id: 'changed_on_delta_humanized',
+          desc: true,
+        },
+      ],
+      filters: getFilters(),
+    });
+  }, [chartFilter]);
+
+  return (
+    <>
+      {sliceCurrentlyEditing && (
+        <PropertiesModal
+          onHide={closeChartEditModal}
+          onSave={handleChartUpdated}
+          show
+          slice={sliceCurrentlyEditing}
+        />
+      )}
+
+      <SubMenu
+        activeChild={chartFilter}
+        name=""
+        // eslint-disable-next-line react/no-children-prop
+        children={[
+          {
+            name: 'Favorite',
+            label: t('Favorite'),
+            onClick: () => setChartFilter('Favorite'),
+          },
+          {
+            name: 'Mine',
+            label: t('Mine'),
+            onClick: () => setChartFilter('Mine'),
+          },
+        ]}
+        buttons={[
+          {
+            name: (
+              <IconContainer>
+                <Icon name="plus-small" /> Chart{' '}
+              </IconContainer>
+            ),
+            buttonStyle: 'tertiary',
+            onClick: () => {
+              // @ts-ignore
+              window.location = '/chart/add';
```

Comment:
If I'm not mistaken in other comments about using `window.location.href` working for this, we can nab this `ts-ignore` and the one below it, too.